rholang/src/main/scala/coop/rchain/rholang/interpreter/Reduce.scala

```diff
@@ -479,15 +479,21 @@ class DebruijnInterpreter[M[_]: Sync: Parallel: _cost](
           p2: Par,
           relopb: (Boolean, Boolean) => Boolean,
           relopi: (Long, Long) => Boolean,
+          relopbi: (BigInt, BigInt) => Boolean,
           relops: (String, String) => Boolean
       ): M[Expr] =
         for {
           v1 <- evalSingleExpr(p1)
           v2 <- evalSingleExpr(p2)
           result <- (v1.exprInstance, v2.exprInstance) match {
-                     case (GBool(b1), GBool(b2))     => GBool(relopb(b1, b2)).pure[M]
-                     case (GInt(i1), GInt(i2))       => GBool(relopi(i1, i2)).pure[M]
-                     case (GString(s1), GString(s2)) => GBool(relops(s1, s2)).pure[M]
+                     case (GBool(b1), GBool(b2)) =>
+                       charge[M](COMPARISON_COST).as(GBool(relopb(b1, b2)))
+                     case (GInt(i1), GInt(i2)) =>
+                       charge[M](COMPARISON_COST).as(GBool(relopi(i1, i2)))
+                     case (GBigInt(bi1), GBigInt(bi2)) =>
+                       charge[M](bigIntComparison(bi1, bi2)).as(GBool(relopbi(bi1, bi2)))
+                     case (GString(s1), GString(s2)) =>
+                       charge[M](COMPARISON_COST).as(GBool(relops(s1, s2)))
```

Comment:
Are these changes related to this PR?

Reply:
Yes, I should have added charging in all cases of `def relop()`. Before the charging was be on top level.

Reply:
I've change location of charging from https://github.com/rchain/rchain/pull/3728/files/b460ffcba2cbef2401634c14088b74e7fea0d122#diff-77048e25dc3a6d74cfc458c5f22f1f475642ca166c79159cc0d781de8c441c5bL570-R635

Reply:
@tgrospic, I deleted charging from here.
https://github.com/rchain/rchain/pull/3728/files/b460ffcba2cbef2401634c14088b74e7fea0d122#diff-77048e25dc3a6d74cfc458c5f22f1f475642ca166c79159cc0d781de8c441c5bL570-R635